interfaces/src/dotty/tools/dotc/interfaces/SimpleReporter.scala

```diff
@@ -0,0 +1,19 @@
+package dotty.tools.dotc.interfaces
+
+import scala.annotation.binaryCompatible
+
+
+/** Report errors, warnings and info messages during the compilation process
+ *
+ * You should implement this interface if you want to handle the diagnostics
+ * returned by the compiler yourself.
+ *
+ * See the method `process` of `dotty.tools.dotc.Driver` for more information.
+ */
+@binaryCompatible
+trait SimpleReporter {
```

Comment:
The spec says that `@binaryCompatible` should not change the public API but a simple trait like that will get a method:
```java
    default public void $init$() {
    }
```
in dotty and in Scala 2.12, but not in Scala 2.11. In Scala 2.12 it will additionally get a method:
```java
    default public static /* synthetic */ void report$(SimpleReporter $this, Diagnostic diag) {
        $this.report(diag);
    }
```

All of these things affect the public API.